src/main/kotlin/io/snyk/plugin/snykcode/SnykCodeBulkFileListener.kt

```diff
@@ -55,13 +68,44 @@ class SnykCodeBulkFileListener : SnykBulkFileListener() {
     }
 
     override fun after(project: Project, virtualFilesAffected: Set<VirtualFile>) {
-        // update .dcignore caches if needed
-        SnykCodeIgnoreInfoHolder.instance.updateIgnoreFileCachesIfAffected(
-            toSnykCodeFileSet(
-                project,
-                virtualFilesAffected
-            )
-        )
+        val filesAffected = toSnykCodeFileSet(project, virtualFilesAffected)
+
+        if (isSnykCodeLSEnabled()) {
+            processAfterForLS(filesAffected, project)
+            return
+        }
+        /* update .dcignore caches if needed */
+        SnykCodeIgnoreInfoHolder.instance.updateIgnoreFileCachesIfAffected(filesAffected)
+    }
+
+    override fun forwardEvents(events: MutableList<out VFileEvent>) {
+        if (!isSnykCodeLSEnabled()) return
+        LanguageServerWrapper.getInstance().ensureLanguageServerInitialized()
+        val languageServer = LanguageServerWrapper.getInstance().languageServer
+        for (event in events) {
+            if (event.file == null || !event.isFromSave) continue
+            val file = event.file!!
+            val activeProject = ProjectUtil.getActiveProject()
+            ProgressManager.getInstance().run(object : Backgroundable(
+                activeProject,
+                "Snyk: forwarding save event to Language Server"
+            ) {
+                override fun run(indicator: ProgressIndicator) {
+                    val param = DidSaveTextDocumentParams(TextDocumentIdentifier(file.url), file.readText())
+                    languageServer.textDocumentService.didSave(param)
+                }
+            })
+        }
+    }
+
+    private fun processAfterForLS(filesAffected: Set<SnykCodeFile>, project: Project) {
+        val cache = getSnykCachedResults(project)?.currentSnykCodeResultsLS ?: return
+        filesAffected.forEach {
+            cache.remove(it)
+            it.virtualFile.getPsiFile(project)?.let { psiFile ->
+                DaemonCodeAnalyzer.getInstance(project).restart(psiFile)
+            }
+        }
```

Comment:
The popup for the `restart` method shows: "_Force re-highlighting for a specific file._". Is `processAfterForLS` main purpose to handle the cache and re analyse?

Since it's being called from `isSnykCodeLSEnabled`, do you think we could rename it to something like `updateCacheAndReanalyze` to make it more obvious? 🤔 Naming is hard, I'm sorry 🙏

Reply:
The restart is repainting the UI, e.g. re-executing the CellRender, so we can grey-out obsolete nodes. Sure, we can rename it. I'll skip the `reanalyze` though, as that's not happening there.

Reply:
It's now named `updateCacheAndUI`.

Reply:
https://github.com/snyk/snyk-intellij-plugin/pull/480